crates/library/std/src/context.fe

```diff
@@ -19,6 +19,60 @@ pub trait Emittable {
   fn emit(self, _ val: OutOfReachMarker);
 }
 
+pub struct CalldataReader {
+    cur_offset: u256
+    len: u256
+
+    pub unsafe fn new(len: u256) -> CalldataReader {
+        return CalldataReader(cur_offset: 0, len)
+    }
+
+    pub fn remainder(self) -> u256 {
+        return self.len - self.cur_offset
+    }
+
+    pub fn advance(mut self, len: u256) -> u256 {
+        self.cur_offset += 32
+        assert self.cur_offset <= self.len
+        return self.cur_offset
+    }
+
+    fn read_n(mut self, len: u256) -> u256 {
+        unsafe {
+            let value: u256 = evm::call_data_load(offset: self.cur_offset)
+            self.advance(len)
+            if len < 32 {
```

Comment:
The if statement here is somewhat costly, is the increased readability/abstration of making all the read_ functions use read_n worth it ??

Reply:
the `read_u256` function could just use `call_data_load` itself and then this if statement can be removed

Reply:
Wouldn't that be redundant and reduce readability ??